### PR TITLE
Gateway structure

### PR DESCRIPTION
**Pull request by LeonardoVieira1630**
# Gateway structure

## All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Description

This pull request introduces a basic gateway structure to the project, developed based on ccip-read. Handlers for functions indicated by ENS interfaces have been implemented, providing an initial integration with it.

## Related Issue
#10 

## Changes
- [x] Gateway Structure: Added the basic gateway structure using ccip-read-server.
- [x] ENS Interfaces Handlers: Implemented handlers for functions indicated by ENS interfaces. The handlers respond to calls and perform specific operations as needed.
- [x] Test File: Introduced a test file to validate gateway calls.

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: packages/gateway/src/gateWay.ts
Comment (pikonha) [inline packages/gateway/src/gateWay.ts:1]: why to call it `gateWay`? perhaps `server` would be a more meaningful name for this file so then after we implement an `index.ts` file that initializes the server, db, and all the other components.

**Review by pikonha — CHANGES_REQUESTED** [review]

File: packages/gateway/README.MD [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- Run yarn to install dependencies.
+
+```
+yarn install
+```
+
+- Run yarn test to execute the tests.
+
+```
+yarn test
```
[the remark this answers is not in the document]
Comment (LeonardoVieira1630) [reply packages/gateway/README.MD:22]: Nice! Will add that.

File: packages/gateway/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "gateway",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
```
[the remark this answers is not in the document]
Comment (LeonardoVieira1630) [reply packages/gateway/package.json:5]: It was set by default as index.js. But yeah, it doesnt exist. I'll change it.

File: packages/gateway/src/gateWay.ts
Comment (LeonardoVieira1630) [reply packages/gateway/src/gateWay.ts:1]: Got you, make sense. This file looks more like the "core" of the gateway. We can call it server, no problem.

**Review by pikonha — APPROVED** [review]
LGTM

**Pull request merged**